File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.kt
class Solution {
    fun solution(name: Array<String>, yearning: IntArray, photo: Array<Array<String>>): IntArray {
        val answer = IntArray(photo.size) { 0 }

        for(i in photo.indices) {
            var sum = 0

            for(j in 0 until photo[i].size) {
                if(photo[i][j] in name) {
                    sum += yearning[name.indexOf(photo[i][j])]
                }
            }

            answer[i] = sum
        }

        return answer
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.swift
func solution(_ name:[String], _ yearning:[Int], _ photo:[[String]]) -> [Int] {
    var answer = [Int]()
    
    for peoples in photo {
        var score = 0
        let longingPeoples = peoples.filter { name.contains($0) }
        
        for longingPeople in longingPeoples {
            if let index = name.firstIndex(of: longingPeople) {
                score += yearning[index]
            }
        }
        
        answer.append(score)
    }
    
    return answer
}